scarlet/component.py

```diff
@@ -159,6 +161,23 @@ def model_to_frame(self, frame=None, model=None):
         return result
 
 
+class Factor(ABC):
+    def __init__(self, *parameters):
+        self._parameters = parameters
+
+    @property
+    def parameters(self):
+        return self._parameters
+
+    @property
+    def bbox(self):
+        return self._bbox
```

Comment:
I'm actually thinking now that each factor should access `model_frame` with an optional `bbox` argument. Otherwise that factor lives in some virtual box but doesn't have access to the properties of that box, e.g. you construct a spectrum but don't know that wavelengths are covered.

I'll refactor...

Reply:
Factors now have `model_frame` and `bbox`